x/capability/keeper/keeper_test.go

```diff
@@ -82,6 +82,7 @@ func (suite *KeeperTestSuite) TestNewCapability() {
 	got, ok := sk.GetCapability(suite.ctx, "transfer")
 	suite.Require().True(ok)
 	suite.Require().Equal(cap, got)
+	suite.Require().True(cap == got, "pointers not equal")
```

Comment:
Indeed `GetCapability` is not reflective of a true object-capability model as you'll get a different memory address every time: https://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/capability/keeper/keeper.go#L250-L256.

My hunch when first reviewing the ADR was why use a memory KVStore? Why not instead just have the Keeper use a primitive map of `module/rev/name => capability`. This would mean that we don't ever encode or decode capabilities, we just keep them ephemerally in the sub-Keeper's state.  You could argue this is an anti-pattern, but some keepers already have ephemeral state (e.g. routers) and maybe this is a model we don't always need to uphold.